fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreBase.java

```diff
@@ -884,6 +884,23 @@ RecordCursor<FDBStoredRecord<M>> scanRecords(@Nullable Tuple low, @Nullable Tupl
                                                  @Nullable byte[] continuation,
                                                  @Nonnull ScanProperties scanProperties);
 
+    /**
+     * Scan a range and return the record primary keys within it.
+     * This method will return a cursor that iterates through all the known primary keys in a range regardless of whether
+     * they consist of a valid record or not. The cursor can be used to find all candidate record primary keys that can be (for example)
+     * scanned for validation. The cursor attempts to account for format versions, split records and inline versions, while
+     * assuming that there may be inconsistencies in the data (The inconsistencies as of right now are missing key-value pairs)
+     *
+     * @param range the scan range
+     * @param continuation a continuation from a previous scan (null if none)
+     * @param scanProperties the scan properties to use (reverse is not supported)
+     *
+     * @return a cursor of Tuples representing the Pks of the records in the range
+     */
+    @Nonnull
+    @API(API.Status.INTERNAL)
+    RecordCursor<Tuple> scanRecordKeys(@Nonnull TupleRange range, @Nullable byte[] continuation, @Nonnull ScanProperties scanProperties);
```

Comment:
I think this should have a default implementation, could be just throwing unsupported, but it's nice to decouple upgrading the library with having to respond to changes.

Reply:
Done.